icu4c/source/common/loclikely.cpp

```diff
@@ -1379,11 +1379,10 @@ ulocimp_getRegionForSupplementalData(const char *localeID, UBool inferRegion,
         rgLen = 0;
```

Comment:
I think we have more problem here. Because the if / else mandate the rgLen to be 6.
But the definition of https://unicode.org/reports/tr35/tr35.html#RegionOverride
said 
"The value is a [unicode_subdivision_id](https://unicode.org/reports/tr35/tr35.html#unicode_subdivision_id) of type “unknown” or “regular”; this consists of a [unicode_region_subtag](https://unicode.org/reports/tr35/tr35.html#unicode_region_subtag) for a regular region (not a macroregion), suffixed either by “zzzz” (case is not significant) to designate the region as a whole, or by a unicode_subdivision_suffix to provide more specificity."

And then
unicode_subdivision_id	= unicode_region_subtag unicode_subdivision_suffix ;
unicode_subdivision_suffix  = alphanum{1,4} ;
 
so in theory unicode_region_subtag could be 2 - 3  and unicode_subdivision_suffix could be 1 - 4 so a valid rgLen should be 3 - 7, not just 6.

For example, if we look at https://github.com/unicode-org/cldr/blob/main/common/supplemental/subdivisions.xml

the following are valid rg value but rgLen != 6
at1
aeaj
afbal

Should we change the if statement to

```
if (U_FAILURE(rgStatus) || rgLen < 3 || rgLen > 7) {
```
or 
```
if (U_FAILURE(rgStatus) || rgLen < 3 || rgLen > 6) {
```
if we assume there are no 3 digits unicode_region_subtag ?

Reply:
>unicode_subdivision_suffix = alphanum{1,4} ;

Ugh.  You're right.  I'll fix this.

I notice in subdivisions.xml, there are some five-letter subdivision codes, but TR35 says one to four letters.  The first two letters are always the same as the region code, so I assume the "subdivision code" is just the characters after the region code?  That is, I'd say `en_GB@rg=USca`, not `en_GB@rg=USusca`?  Or, to take a five-letter example, `en_GB@rg=AUnsw` and not `en_GB@AUaunsw`?

Reply:
Yes, en_GB@rg=USca (or equivalently en-GB-u-rg-usca)
It is possible in the future to have 3-digit regions, so 7 would be safer as long as you are touching the code.

Reply:
That means actually parsing the string, which is something we've avoided up until now.  I was kind of hoping we could avoid going there until we really knew we needed it, but I've gone back, added a test, and done a really cheap and hacky version of paring the string to get the region code.  Let me know if I need to go to greater lengths.